openpype/lib/avalon_context.py

```diff
@@ -261,6 +262,200 @@ def get_hierarchy(asset_name=None):
+    works with handles as they're extending frame start/end.
+
+    ## Example
+    Data from asset document:
+    - frame start: 1001
+    - frame end: 1100
+    - handle start: 10
+    - handle end: 10
+
+    1. Project has stored that handles "Extend"
+    - frame range with handles: 991 - 1110 (120 frames)
+    - frame range without handles: 1001 - 1100 (100 frames)
+
+    2. Project has stored that handles "Reduce"
+    - frame range with handles: 1001 - 1100 (100 frames)
+    - frame range without handles: 1011 - 1090 (80 frames)
+
+
+    Class unifies access too `frame_start`, `fram_end`, `handle_frame_start`
+    and `handle_frame_end`.
+
+    Keeps tray about current state of handles if are extending of reducing
+    frame range. If you want to get data to store please set if handles
+    should be extended using `change_handles_extend` and then
+    `real_frame_start` and `real_frame_end`.
+    """
+
+    def __init__(
+        self,
+        frame_start,
+        frame_end,
+        handle_start,
+        handle_end,
+        handles_extend=None
+    ):
+        if handles_extend is None:
+            handles_extend = True
+        self.frame_start = frame_start
+        self.frame_end = frame_end
+        self.handle_start = handle_start or 0
+        self.handle_end = handle_end or 0
+        self._handles_extend = True
+        self.change_handles_extend(handles_extend)
+
+        self._log = None
+
+    @property
+    def log(self):
+        if self._log is None:
+            self._log = PypeLogger.get_logger(self.__class__.__name__)
+        return self._log
+
+    def __str__(self):
+        return "< {}: ({}) {}-{} ({}) >".format(
+            self.__class__.__name__,
+            self.handle_frame_start,
+            self.frame_start,
+            self.frame_end,
+            self.handle_frame_end
+        )
+
+    @property
+    def handle_frame_start(self):
+        """Frame start with handles."""
+        # QUESTION should we reduce handles if they're lower than '0'?
+        return self.frame_start - self.handle_start
+
+    @property
+    def handle_frame_end(self):
+        """Frame end with handles."""
+        return self.frame_end + self.handle_end
+
+    @property
+    def real_frame_start(self):
+        """Frame start that would be stored in project or asset doc."""
+        if self._handles_extend:
+            return self.frame_start
+        return self.handle_frame_start
+
+    @property
+    def real_frame_end(self):
+        """Frame end that would be stored in project or asset doc."""
+        if self._handles_extend:
+            return self.frame_end
+        return self.handle_frame_end
+
+    @property
+    def frame_range_without_handles(self):
+        """Frame range without handles."""
+        return (self.frame_end - self.frame_start) + 1
+
+    @property
+    def frame_range(self):
+        """Frame range with handles."""
+        return (self.handle_frame_end - self.handle_frame_start) + 1
```

Comment:
To me these aren't frame ranges but `durations` (or "number of frames"). A frame range to me consists of two values, like a start and end.

The loader currently shows these as **Duration** (excluding the handles):
![afbeelding](https://user-images.githubusercontent.com/2439881/155882365-3d6fa37e-80a6-4244-9fce-2cc693c03cb9.png)

Reply:
Agree. Other attributes are clear?

Reply:
I had to re-read the class multiple times and I am still not sure I fully grasp the variables. So at first sight. No.

Let me read up again coming week when I have some time to let this sink in and see if I can come up with better input.